be/src/storage/tablet.cpp

```diff
@@ -836,9 +836,19 @@ Status Tablet::_capture_consistent_rowsets_unlocked(const std::vector<Version>&
     return Status::OK();
 }
 
-// TODO(lingbin): what is the difference between version_for_delete_predicate() and
-// version_for_load_deletion()? should at least leave a comment
 bool Tablet::version_for_delete_predicate(const Version& version) {
+    // we use try/catch-block to lock _meta_lock in share mode
+    // It is because in some case, _meta_lock may has been acquire
+    // this the current thread in shared/exclusive mode. It may
+    // be crash if we acquire share lock again. We just do nothing
+    // in this case.
+    try {
+        std::shared_lock rlock(get_header_lock());
+    } catch (const std::exception& e) {}
```

Comment:
This lock will be released after this try block, are you sure that this will fix the bug?

Reply:
Yes, it is a problem, maybe we can use get_header_lock().lock_shared() instead. But i still reviewing implementation is reasonable or not. I will fix it later

Reply:
updated